include/ginkgo/core/matrix/dense.hpp

```diff
@@ -161,6 +161,26 @@ class Dense
         return (*other).create_with_same_config();
     }
 
+    /**
+     * Creates a Dense matrix with the same type and executor as another Dense
+     * matrix but a different size.
+     *
+     * @param other  The other matrix whose type we target.
+     * @param exec  The executor of the new matrix.
+     * @param size  The size of the new matrix.
+     * @param stride  The stride of the new matrix.
+     *
+     * @returns a Dense matrix with the type of other.
+     */
+    static std::unique_ptr<Dense> create_with_type_of(
+        const Dense *other, std::shared_ptr<const Executor> exec,
+        const dim<2> &size = dim<2>{}, size_type stride = 0)
+    {
+        auto used_stride = stride == 0 ? size[1] : stride;
```

Comment:
From the interface documentation view, this might be a bit confusing, how about having two overloads (one with size = dim<2>{}, one with all parameters, since value 0 as a special indicator for the stride to me is less clear than indicating this by the presence/absense of a parameter.

Reply:
I added an extra overload which takes all parameters whereas the other ones still uses a default `dim<2>{}` if needed but takes no stride.

Reply:
That's exactly what I meant, thanks :)